tests/internal/test_remoteconfig.py

```diff
@@ -132,3 +132,9 @@ def _reload_features(self, metadata, features):
         sleep(0.2)
         mock_send_request.assert_called_once()
         assert callback.features == {"asm": {"enabled": True}}
+
+
+def test_remoteconfig_semver():
+    version = RemoteConfigClient()._client_tracer["tracer_version"]
+    if "rc" in version:
+        assert "-rc" in version, version
```

Comment:
instead check the semver with "in", it is possible to use a regex (https://regex101.com/r/vkijKf/1/) to be sure the version is semver complance, what do you think?

Reply:
Yes please, I don't think this change actually makes it semver compliant? since the `+` isn't an allowed character in semver.

Reply:
> instead check the semver with "in", it is possible to use a regex (https://regex101.com/r/vkijKf/1/) to be sure the version is semver complance, what do you think?

This is based on the assumption that we either generate a version of the form `x.y.z` (for release builds) or something of the form `x.y.zrc...`. So to keep the logic as simple as possible I thought of just handling the `rc` case. Do you reckon we should be more thorough here to cover cases that I've potentially overlooked?